### PR TITLE
Ignore case when checking zip extension

### LINKED ISSUE #6491: Crashing when entering (sometimes) my Images folder
### Description

Sometimes if I enter my Images folder it will crash with the "something went wrong" notification and this on the debug.log:

> 2021-10-13 21:09:01.8454|ERROR|AppUnhandledException|Object reference not set to an instance of an object.
> 	System.NullReferenceException: Object reference not set to an instance of an object.
>    at Files.Views.LayoutModes.GridViewBrowser.GridViewBrowser_obj27_Bindings.Update_LoadFileIcon(Boolean, Int32) + 0x43
>    at Files.Views.LayoutModes.GridViewBrowser.GridViewBrowser_obj27_Bindings.GridViewBrowser_obj27_BindingsTracking.PropertyChanged_(Object, PropertyChangedEventArgs) + 0x469
>    at System.Action`2.InvokeMulticastThunk(T1, T2) + 0x4e
>    at System.EventHandler`1.Invoke(Object, TEventArgs) + 0x2e
>    at Microsoft.Toolkit.Mvvm.ComponentModel.ObservableObject.SetProperty[T](T&, T, String) + 0x9c
>    at Files.Filesystem.ListedItem.Img_ImageOpened(Object, RoutedEventArgs) + 0x55
>    at Microsoft.Toolkit.Uwp.UI.Controls.InAppNotificationOpeningEventHandler.Invoke(Object, InAppNotificationOpeningEventArgs) + 0x2e
>    at __Interop.Intrinsics.HasThisCall__23[TArg0](Object, IntPtr, Object, TArg0) + 0x36
>    at Files!<BaseAddress>+0x12b1f30

debug_fulltrust.log:

> 2021-10-13 21:10:14.4828|WARN|ParseFileOperationAsync|Devuelto error HRESULT E_FAIL de una llamada a un componente COM.
> 	System.Runtime.InteropServices.COMException (0x80004005): Devuelto error HRESULT E_FAIL de una llamada a un componente COM.
>    en Vanara.PInvoke.Shell32.IShellLinkW.Resolve(HWND hwnd, SLR_FLAGS fFlags)
>    en Vanara.Windows.Shell.ShellLink.LoadAndResolve(String linkFile, SLR_FLAGS resolveFlags, HWND hWin, UInt16 timeOut)
>    en FilesFullTrust.MessageHandlers.FileOperationsHandler.<ParseFileOperationAsync>d__4.MoveNext()
> 2021-10-13 21:10:14.5378|INFO|Connection_RequestReceived|Argument: FileOperation
> 2021-10-13 21:10:14.5488|WARN|ParseFileOperationAsync|El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
> 	System.IO.FileNotFoundException: El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
>    en Vanara.PInvoke.Shell32.ShellUtil.GetShellItemForPath(String path)
>    en Vanara.Windows.Shell.ShellLink.get_Target()
>    en FilesFullTrust.MessageHandlers.FileOperationsHandler.<ParseFileOperationAsync>d__4.MoveNext()

On the end of the post I will add the full logs, tough there is nothing else relevant I think.

### Steps To Reproduce

1. Go to a folder that has a lot of content (I guess... my Images folder has 3552 items)
2. If it's not crashing, open a new tab and try again.

It's not crashing all the times, but eventually it will crash (more or less at the third time, but I got this on the first time I opened Images folder after installing last version), at least in my testings.

### Expected behavior

Should not crash

### Files Version

V2.0.19.0

### Windows Version

Windows 10 21H1 10.0.19043.1288

### Relevant Assets

[debug.log](https://github.com/files-community/Files/files/7340743/debug.log)
[debug_fulltrust.log](https://github.com/files-community/Files/files/7340744/debug_fulltrust.log)

In my testings:
image

### LINKED ISSUE #6495: Invocation of any zip file whose extension is capitalised is not operational.
### Description

image

### Steps To Reproduce

Invoke any zip file whose extension is capitalised.

### Expected behavior

This software should either support management (observation and decompression) of capitalised ZIP files.

### Files Version

v2.0.13.0

### Windows Version

Windows 11 22471.1000

### Relevant Assets

[debug_fulltrust.log](http://github.com/files-community/Files/files/7341842/debug_fulltrust.log)
[debug.log](http://github.com/files-community/Files/files/7341843/debug.log)

### PR DESCRIPTION
**Pull request by gave92**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #6495
- Closes #6491
- Test fix for #6605
- Fixes 1152067677u
- Fixes 2989330824u
- Fixes 2009482836u

**Details of Changes**
Add details of changes here.
- Added try-catch when extracting files
- Compare zip extension ignoring case
- Regenerate file tags json file if invalid

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

File: Files/Filesystem/ListedItem.cs [hunk 1]
```diff
@@ -204,11 +204,14 @@ private void Img_ImageOpened(object sender, Windows.UI.Xaml.RoutedEventArgs e)
...
-                    LoadDefaultIcon = false;
-                    LoadWebShortcutGlyph = false;
+                    Common.Extensions.IgnoreExceptions(() =>
+                    {
+                        LoadFileIcon = true;
+                        PlaceholderDefaultIcon = null;
+                        NeedsPlaceholderGlyph = false;
+                        LoadDefaultIcon = false;
+                        LoadWebShortcutGlyph = false;
+                    }, App.Logger); // 2009482836u
```
Comment (d2dyno1) [inline Files/Filesystem/ListedItem.cs:214]: Why would it fail?

File: Files/Services/Implementation/FileTagsSettingsService.cs [hunk 2]
```diff
@@ -33,6 +33,11 @@ public IList<FileTag> FileTagList
 
         public FileTag GetTagById(string uid)
         {
+            if (FileTagList.Any(x => x.Uid == null))
```
Comment (d2dyno1) [inline Files/Services/Implementation/FileTagsSettingsService.cs:36]: .IsNullOrEmpty() ?

File: Files/Filesystem/ListedItem.cs (hunk 1 above)
Comment (gave92) [reply Files/Filesystem/ListedItem.cs:214]: Good question, I do not know for sure.The referenced stacktrace has always been there but we had a try-catch around it.

File: Files/Services/Implementation/FileTagsSettingsService.cs (hunk 2 above)
Comment (gave92) [reply Files/Services/Implementation/FileTagsSettingsService.cs:36]: I need to check if Filelist contains any invalid items (items with a null ID).
I'm considering removing this as I'm not certain solves the related issue.

**Review by d2dyno1 — APPROVED** [review]
lgtm

**Comment by yaichenbaum** [comment]
Thank you!

**Pull request merged**